src/shogun/kernel/SigmoidKernel.cpp

```diff
@@ -57,9 +57,13 @@ void CSigmoidKernel::init()
 	gamma = 0.0;
 	coef0 = 0.0;
 
-	SG_ADD(
-	    &gamma, "gamma", "Scaler for the dot product.",
-	    ParameterProperties::HYPER | ParameterProperties::AUTO,
-	    std::make_shared<params::GammaFeatureNumberInit>(this));
+	sg_add(&gamma, "gamma", "Scaler for the dot product.",
+		   ParameterProperties::HYPER | ParameterProperties::AUTO,
+		   std::make_shared<params::GammaFeatureNumberInit>(this));
```

Comment:
would it be possible to move the `make_shared` into the method rather than having to write it everytime a parameter is registered?

Reply:
I am not sure how that would work though. We could pass the type `GammaFeatureNumberInit`, i.e. `sg_add<params::GammaFeatureNumberInit>(&gamma, "gamma", "Scaler for the dot product.", ParameterProperties::HYPER | ParameterProperties::AUTO);`, but I am not sure that is nicer?

Reply:
Wouldn't that require too many classes?

Reply:
in what sense? you mean for template instantiations?

Reply:
Will we have `params::GammaFeatureNumberInit`, `params::AlphaFeatureNumberInit`, etc?

Reply:
yes, but the templated method can check if the passed type is derived from the abstract class (I think its `AbstractInit`). not sure if that's what you mean?